papis/commands/doctor.py

```diff
@@ -0,0 +1,374 @@
+@papis.cli.query_option()
+@papis.cli.sort_option()
+@click.option("-t", "--checks", "_checks",
+              default=lambda: papis.config.getlist("doctor-default-checks"),
+              multiple=True,
+              help=("Checks to run on every document, possible values: {}"
+                    .format(", ".join(registered_checks_names()))))
+@click.option("--json", "_json",
+              default=False, is_flag=True,
+              help="Output the results in json format")
+@click.option("--fix",
+              default=False, is_flag=True,
+              help="Auto fix the errors with the auto fixer mechanism")
+@click.option("-s", "--suggest",
+              default=False, is_flag=True,
+              help="Suggest commands to be run for resolution")
+@click.option("-e", "--explain",
+              default=False, is_flag=True,
+              help="Give a short message for the reason of the error")
+@click.option("--edit",
+              default=False, is_flag=True,
+              help="Edit every file with the edit command.")
+@papis.cli.all_option()
+@papis.cli.doc_folder_option()
+def cli(query: str,
+        doc_folder: str,
+        sort_field: Optional[str],
+        sort_reverse: bool,
+        _all: bool,
+        fix: bool,
+        edit: bool,
+        explain: bool,
+        _checks: List[str],
+        _json: bool,
+        suggest: bool) -> None:
+    """Check for common problems in documents"""
+
+    # handle doc_folder
+    if doc_folder:
+        documents = [papis.document.from_folder(doc_folder)]
+    else:
+        documents = papis.database.get().query(query)
+
+    if not _all:
+        documents = [doc for doc in papis.pick.pick_doc(documents) if doc]
+
+    if sort_field:
+        documents = papis.document.sort(documents, sort_field, sort_reverse)
+
+    if not documents:
+        logger.warning(papis.strings.no_documents_retrieved_message)
+        return
+
+    logger.debug("Running checks: %s", _checks)
+
+    errors = []  # type: List[Error]
+    for doc in documents:
+        errors += run(doc, _checks)
+
+    if errors:
+        logger.warning("%s errors found", len(errors))
+
+    if _json:
+        print(json.dumps(list(map(lambda e:
+                                  dict(msg=e.payload,
+                                       path=e.path,
+                                       name=e.name,
+                                       suggestion=e.suggestion_cmd),
+                                  errors))))
+        return
+
+    for error in errors:
+        print("{e.name}\t{e.payload}\t{e.path}".format(e=error))
+        if explain:
+            print("\tReason: {}"
+                  .format(error.msg))
+        if suggest:
+            print("\tSuggestion: {}"
+                  .format(error.suggestion_cmd))
+        if fix:
+            logger.warning("Fixing...")
+            error.fix_action()
+        if edit:
+            input("Press any key to edit...")
+            if error.doc:
```

Comment:
```suggestion
            if error.doc:
            	input("Press any key to edit...")
```
? Or add a message that it got skipped? Seems weird to press a key to edit and nothing happens

Reply:
I only made doc to be optional to be kind of future-proof maybe with the explorer of documents, so practically there must be always a document, but you're right, like this is better.